src/main/java/com/minecolonies/network/messages/BuildToolPlaceMessage.java

```diff
@@ -140,34 +133,58 @@ public IMessage onMessage(@NotNull BuildToolPlaceMessage message, @NotNull Messa
      */
     private static void handleHut(@NotNull World world, @NotNull EntityPlayer player, String hut, String style, int rotation, @NotNull BlockPos buildPos)
     {
-        if (Schematics.getStylesForHut(hut) == null)
+        if (Structures.getStylesForHut(hut) == null)
         {
             Log.getLogger().error("No record of hut: " + hut);
             return;
         }
+        Colony tempColony = ColonyManager.getClosestColony(world, buildPos);
+        //if (tempColony != null && !tempColony.getPermissions().hasPermission(player, Permissions.Action.MANAGE_HUTS))
+        {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1199)